ui/plugins/pinboard.js

```diff
@@ -245,5 +249,36 @@ treeherder.controller('PinboardCtrl', [
         $scope.pinnedJobs = thPinboard.pinnedJobs;
         $scope.relatedBugs = thPinboard.relatedBugs;
 
+        var getRevisionTips = function(projectName, list) {
+            list.splice(0, list.length);
+            ThResultSetModel.getResultSets(projectName).then(function(response) {
+                var resultsets = response.data.results;
+                resultsets.forEach(function(revisionSet) {
+                    list.push({
+                        revision: revisionSet.revision,
+                        author: revisionSet.author
+                    });
+                });
+            });
+        };
+
+        $scope.updateOriginalgRevisionTips = function() {
+            getRevisionTips(repoName, $scope.originalTipList);
+        };
+
+        $scope.updateNewRevisionTips = function() {
+            getRevisionTips(repoName, $scope.newTipList);
+        };
+
+        $scope.updateOriginalgRevisionTips();
+        $scope.updateNewRevisionTips();
+
+        $scope.getOriginalTipRevision = function(tip) {
+            $scope.originalRevision = tip;
```

Comment:
no need to store ``originalRevision`` in ``$scope`` because you don't use it anywhere else.